### PR TITLE
[FIX/ADD] configuration options

### PR DESCRIPTION
**Pull request by nilshamerlinck**
1) https://github.com/OCA/queue/commit/652911b9e7ecdcbb76dcfeedd752b23b4ed64093 re-introduced an harcoded `http://localhost` ;-)
2) Port of OCA/connector#248

### CODE REVIEW

**Comment by guewen** [comment]
> 652911b re-introduced an harcoded http://localhost ;-)

Oops, sorry :expressionless:

**Review by guewen — APPROVED** [review]
LGTM

**Comment by guewen** [comment]
Only a few lint from travis to fix:
```
queue_job/jobrunner/runner.py:29:80: E501 line too long (96 > 79 characters)
queue_job/jobrunner/runner.py:33:80: E501 line too long (94 > 79 characters)
queue_job/jobrunner/runner.py:34:80: E501 line too long (89 > 79 characters)
```

File: queue_job/jobrunner/runner.py [hunk 1]
```diff
@@ -179,21 +183,38 @@ def _odoo_now():
     return _datetime_to_epoch(dt)
 
 
+def _connection_info_for(db_name):
+    db_or_uri, connection_info = odoo.sql_db.connection_info_for(db_name)
+
+    for p in ('host', 'port'):
```
Comment (simahawk) [inline queue_job/jobrunner/runner.py:192]: `s/p/[key|param]`

**Review by simahawk — APPROVED** [review]
LG, I'd replace a var name, but that's it :)
tnx for the change!

File: queue_job/jobrunner/runner.py (hunk 1 above)
Comment (nilshamerlinck) [reply queue_job/jobrunner/runner.py:192]: If you don't mind, I will keep it as it to keep it consistent with the same piece of code that is in the other versions of the module.

Comment (simahawk) [reply queue_job/jobrunner/runner.py:192]: yeah, sorry, I missed the original code :sweat_smile:  :+1:

**Comment by whulshof** [comment]
Could this merge cause queue to not start anymore? I took the latest 10.0 merge and deployed it in my test-environment and now no jobs will start anymore. In the log the job runner gives the usual acknowledgement it runs on the db etc.
Should I change any configuration to get it running again? Which I would think is not to be expected, but anyhow.

**Comment by nilshamerlinck** [comment]
Hi @whulshof, the PR for 10.0 is https://github.com/OCA/queue/pull/77

Any error message? Are you using a connection pooler (pgbouncer)?

**Comment by whulshof** [comment]
No, no error messages, although log in DEBUG level. And no connection pooler either.


Met vriendelijke groet/Kind regards,

Magnus Red

Willem Hulshof

+31 88 696 60 00 | +31 651 243 710 | www.magnus.nl<http://www.magnus.nl/red/>

[id:4FD331F6-E3AE-4679-826C-9D6015959C97]

Van: Nils Hamerlinck <notifications@github.com>
Beantwoorden - Aan: OCA/queue <reply@reply.github.com>
Datum: donderdag 2 augustus 2018 om 14:04
Aan: OCA/queue <queue@noreply.github.com>
CC: Willem Hulshof <w.hulshof@magnus.nl>, Mention <mention@noreply.github.com>
Onderwerp: Re: [OCA/queue] [FIX/ADD] configuration options (#86)


Hi @whulshof<https://github.com/whulshof>, the PR for 10.0 is #77<https://github.com/OCA/queue/pull/77>

Any error message? Are you using a connection pooler (pgbouncer)?

**Comment by sbidoul** [comment]
A common cause for jobs not starting is jobs in remaining in state `enqueued` or `started` when the server has not been shut down properly.

**Comment by whulshof** [comment]
I know that, so I requeued or set to done all pending, started en enqueued jobs already. To no avail.
But do I have to do any configuration change as the result of Nils’ PR?


Met vriendelijke groet/Kind regards,

Magnus Red

Willem Hulshof

+31 88 696 60 00 | +31 651 243 710 | www.magnus.nl<http://www.magnus.nl/red/>

[id:4FD331F6-E3AE-4679-826C-9D6015959C97]

Van: "Stéphane Bidoul (ACSONE)" <notifications@github.com>
Beantwoorden - Aan: OCA/queue <reply@reply.github.com>
Datum: donderdag 2 augustus 2018 om 14:16
Aan: OCA/queue <queue@noreply.github.com>
CC: Willem Hulshof <w.hulshof@magnus.nl>, Mention <mention@noreply.github.com>
Onderwerp: Re: [OCA/queue] [FIX/ADD] configuration options (#86)


A common cause for jobs not starting is jobs in remaining in state enqueued or started when the server has not been shut down properly.

**Comment by lmignon** [comment]
@whulshof do you have more than 1 database visible by your Odoo instance? If it's the case, can you retry with only 1 database visible by odoo?

**Comment by whulshof** [comment]
Yes, i’ll do right away


Met vriendelijke groet/Kind regards,

Magnus Red

Willem Hulshof

+31 88 696 60 00 | +31 651 243 710 | www.magnus.nl<http://www.magnus.nl/red/>

[id:4FD331F6-E3AE-4679-826C-9D6015959C97]

Van: "Laurent Mignon (ACSONE)" <notifications@github.com>
Beantwoorden - Aan: OCA/queue <reply@reply.github.com>
Datum: donderdag 2 augustus 2018 om 14:40
Aan: OCA/queue <queue@noreply.github.com>
CC: Willem Hulshof <w.hulshof@magnus.nl>, Mention <mention@noreply.github.com>
Onderwerp: Re: [OCA/queue] [FIX/ADD] configuration options (#86)


@whulshof<https://github.com/whulshof> do you have more than 1 database visible by your Odoo instance? If it's the case, can you retry with only 1 database visible by odoo?

**Comment by whulshof** [comment]
This seems to be doing the trick! Is there a possibility to keep the job queue working for more than 1 database, like before? If yes, how?


Met vriendelijke groet/Kind regards,

Magnus Red

Willem Hulshof

+31 88 696 60 00 | +31 651 243 710 | www.magnus.nl<http://www.magnus.nl/red/>

[id:4FD331F6-E3AE-4679-826C-9D6015959C97]

Van: "Laurent Mignon (ACSONE)" <notifications@github.com>
Beantwoorden - Aan: OCA/queue <reply@reply.github.com>
Datum: donderdag 2 augustus 2018 om 14:40
Aan: OCA/queue <queue@noreply.github.com>
CC: Willem Hulshof <w.hulshof@magnus.nl>, Mention <mention@noreply.github.com>
Onderwerp: Re: [OCA/queue] [FIX/ADD] configuration options (#86)


@whulshof<https://github.com/whulshof> do you have more than 1 database visible by your Odoo instance? If it's the case, can you retry with only 1 database visible by odoo?

**Pull request merged**